protocol/nats_jetstream/v2/message.go

```diff
@@ -25,7 +38,15 @@ type Message struct {
 // NewMessage wraps an *nats.Msg in a binding.Message.
 // The returned message *can* be read several times safely
 func NewMessage(msg *nats.Msg) *Message {
-	return &Message{Msg: msg, encoding: binding.EncodingStructured}
+	encoding := binding.EncodingStructured
```

Comment:
I thought this PR is `binary` mode only?

Reply:
Structured mode was supported before this change.  This PR is to allow the choice of binary mode side-by-side with the already existing structured mode.